azext_iot/operations/_mqtt.py

```diff
@@ -45,12 +46,20 @@ def message_handler(self, message):
 
         message_properties.update(message.custom_properties)
 
+        if message.data and "content_encoding" in message_properties:
```

Comment:
"content_encoding" in message_properties will return True if that key is in the dictionary, not if it also has a truthy value.

Reply:
Got it, I am going to replace this by a message_properties.get("content_encoding") != None. That will ensure the key exists and has a value

Reply:
https://github.com/Azure/azure-iot-cli-extension/pull/362/files#diff-51346fa92f494b0aa5b1694cae7bd0334bb53a95fd92a9d715a4ddba323871fcR44 will filter out properties that don't have values allowing this to work